Add Bday Event + Fixups and Changes

Berthalamew opened the pull request:
-implement bday event
-cleanup event code
-improve tag defs
-add option to open new menus on the fly in dev settings menu
-add option to force an event type in the dev menu
-remove all powerups and vehicles from infection
-Fix highresfix

Berthalamew committed:
Bday event update

-implement bday event
-cleanup event code
-improve tag defs
-add option to open new menus on the fly in dev settings menu
-add option to force an event type in the dev menu
-remove all powerups and vehicles from infection

Berthalamew committed:
Fix highresfix

nukeulater commented on xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.h:
```diff
@@ -7,12 +13,194 @@ namespace SpecialEvents
+		float scale;
+	};
+
+	const s_event_scenery_placement lockout_scen_places[HALLOWEEN_LOCKOUT_OBJECT_COUNT] {
```
consider moving these to files at some point

nukeulater commented on xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:
```diff
@@ -337,346 +310,125 @@ namespace SpecialEvents
 			auto pump = tags::get_tag<blam_tag::tag_group_type::scenery, s_scenery_group_definition>(pump_datum, true);
 			auto pump_hmlt = tags::get_tag<blam_tag::tag_group_type::model, s_model_group_definition>(pump->objectTag.model.TagIndex, true);
+			
+			char* mapName = Memory::GetAddress<char*>(0x47CF0C);
```
I think this can be read using tag interface helpers, from the cache header

Berthalamew committed:
Cleanup and Crash Fix

- Cleanup PlayerRep code
- Fix crash since event code tries to access the new reps before they're created

nukeulater commented on xlive/H2MOD/Modules/Input/PlayerControl.cpp:
```diff
@@ -5,27 +5,27 @@
 void PlayerControl::ApplyHooks()
 {
-	return; // TODO FIXME usercall convention functions, bad hooks
+	//return; // TODO FIXME usercall convention functions, bad hooks
```
this can be removed

nukeulater commented on xlive/H2MOD/Variants/Infection/Infection.cpp:
```diff
@@ -134,50 +134,46 @@ void Infection::InitHost() {
+	for each (auto itemcollection in itemcollections)
+		{
+			std::string item_name = tags::get_tag_name(itemcollection.first);
+			if (strstr(item_name.c_str(), "multiplayer\\powerups") != nullptr ||
```
keep it consistent, use c++'s std::string helpers (`find`) here instead of `strstr`

Berthalamew committed:
Requested changes + cleanup

- add camera struct and cleanup code relating to it
- move render_camera_build_projection_hook to engine specific folder
- move object placements for events to seperate files
- remove test code in MetaExtender
- fix issue in playerep code from previous commit
